### ISSUE TITLE
Inheritance from a Generic Type

### ISSUE THREAD

**Issue by KeithPepin** [nicolasdao/graphql-s2s#35]
@nicolasdao thanks for the great library. I've been enjoying how it simplified some recent code I developed at work. I do have a quick question regarding inheritance and generics though. Am I wrong in thinking something like this should work? (currently getting errors when I try with graphql.js):

```
enum MyEnum {
  option1
  option2
}

type BaseGeneric<T> {
  enumneratedThing: T!
  otherProperty: String
}

type FinalType inherits BaseGeneric<MyEnum> {
  someOtherProperty: String!
}
```

Which would ideally transpile the FinalType into something like this:

```
type FinalType {
  enumeratedThing: MyEnum!
  otherProperty: String
  someOtherProperty: String!
}
```

If this currently isn't supported I have this use case, and would love to see this sort of feature added.

**Comment by nicolasdao** [comment]
Thanks a lot for reporting this issue @KeithPepin and thanks for detailing your example. We have enough details to work on a fix. 

We'll implement that feature asap.

**Comment by KeithPepin** [comment]
Wonderful @nicolasdao ! I'll look forward to when it is released.

**Comment by nicolasdao** [comment]
Hi @KeithPepin, 

I've just released version 0.20.0 which fixes this issue. 

Thanks for reporting that issue.

Cheers,

Nic

**Comment by KeithPepin** [comment]
Awesome, thanks @nicolasdao !
